### ISSUE TITLE
Adding a comment next to `#![feature(ptr_metadata,)]` silences failing doctests

### ISSUE THREAD

**Issue by raldone01** [rust-lang/rust#134221]
### Reproduce

https://github.com/ink-feather-org/trait_cast_rs/tree/80a0a0994dfdbf5e8b6978879ce3205188401df1

#### 1:

```shell
git clone https://github.com/ink-feather-org/trait_cast_rs.git
git checkout 80a0a0994dfdbf5e8b6978879ce3205188401df1
cargo test --all-features
```

All tests pass!

#### 2:

Remove the comment in line 43 of the `README.md`:

```rust
#![feature(
  ptr_metadata, // <- Remove this comment
)]
use trait_cast_rs::{
  make_trait_castable, TraitcastableAny, TraitcastableAnyInfra, TraitcastableAnyInfraExt,
};
#[make_trait_castable(Print)]
struct Source(i32);
...
```

Next run the tests again:

```shell
cargo test --all-features
```

Oh no! The test fails with `E0119`.

What is going on?!? I did not expect the comment breaking anything.

### Meta
`rustc --version --verbose`:
```
rustc 1.85.0-nightly (21fe748be 2024-12-11)
binary: rustc
commit-hash: 21fe748be15271ea5804e0507cd699b675efe038
commit-date: 2024-12-11
host: x86_64-unknown-linux-gnu
release: 1.85.0-nightly
LLVM version: 19.1.5
```

**Comment by cyrgani** [comment]
The bug appears to be not that your test fails, but rather that it suceeds when you add the comment.
Here is a reduced example:
```rust
/*!```rust
#![feature(
  foo, //
)]
```
*/
```
This as a sole `lib.rs` should fail after `cargo test` because `foo` isn't a feature, but it silently suceeds with the comment. Removing it causes the test to fail as expected.

**Comment by jieyouxu** [comment]
Would be good to get a more precise bisection and a more self-contained MCVE, looks not ideal.

**Comment by jieyouxu** [comment]
Wait on hold, @cyrgani to double-check, do you mean this as a *doctest*?

**Comment by jieyouxu** [comment]
searched nightlies: from nightly-2024-07-20 to nightly-2024-08-30
regressed nightly: nightly-2024-08-15
searched commit range: https://github.com/rust-lang/rust/compare/80eb5a8e910e5185d47cdefe3732d839c78a5e7e...13a52890dde8cfeb95069d77c223ac37c0cf3a46
regressed commit: https://github.com/rust-lang/rust/commit/e5b3e68abf170556b9d56c6f9028318e53c9f06b

<details>
<summary>bisected with <a href='https://github.com/rust-lang/cargo-bisect-rustc'>cargo-bisect-rustc</a> v0.6.9</summary>


Host triple: x86_64-unknown-linux-gnu
Reproduce with:
```bash
cargo bisect-rustc --script=./check.sh --start=1.81.0 --end=2024-08-30 --regress=success
```
</details>

**Comment by jieyouxu** [comment]
cc @GuillaumeGomez, do you know if this is intended behavior change?

**Comment by raldone01** [comment]
> Intended behavior change

Good one. 🫠

It's true my doctest was broken but the `//` in the `feature` expression silenced the failure.
If I had kept the comment I would have never noticed that the doctest was failing.

**Comment by onestacked** [comment]
Great now I know how to get all my doctests to pass!

**Comment by cyrgani** [comment]
Note that there seem to be 2 underlying problems:
1. Any doctest starting with a malformed inner attribute is silently accepted, even just
```rust
//! ```rust
//! #![
//! ```
```
2. I guess that line breaks inside the attribute are removed before comments are removed, so adding the comment creates a malformed attribute and 1. applies.

**Comment by GuillaumeGomez** [comment]
No it's not an intended behaviour change. ^^'

Gonna check what's going on.

**Comment by GuillaumeGomez** [comment]
Opened https://github.com/rust-lang/rust/pull/134260.
